ha/resource/dynamic_fid_service_ra.py

```diff
@@ -0,0 +1,266 @@
+        It map clone to service instance on node.
+        </longdesc>
+        <shortdesc lang="en">Systemd wrapper agent</shortdesc>
+        <parameters>
+        <parameter name="service" required="0">
+        <longdesc lang="en"> Service name to manage systemd </longdesc>
+        <shortdesc lang="en"> Systemd service </shortdesc>
+        <content type="string"/>
+        </parameter>
+        <parameter name="fid_service_name" required="0">
+        <longdesc lang="en"> Fid service name used in mapping </longdesc>
+        <shortdesc lang="en"> Systemd service </shortdesc>
+        <content type="string"/>
+        </parameter>
+        </parameters>
+        <actions>
+        <action name="start"        timeout="40s" />
+        <action name="stop"         timeout="40s" />
+        <action name="monitor"      timeout="40s" interval="60s" depth="0" />
+        <action name="meta-data"    timeout="4s" />
+        </actions>
+        </resource-agent>
+        """
+        sys.stdout.write(env)
+        return const.OCF_SUCCESS
+
+    def start(self) -> int:
+        """
+        Start service and provide output.
+
+        Command to start service:
+            $ systemctl reset-failed service
+            $ systemctl start service
+
+        Returns:
+            int: Return as per service status.
+                active: return const.OCF_SUCCESS.
+                unknown: Wait till timeout.
+                failed or timeout will cause failover or moved to Stopped state.
+        """
+        service = self._get_systemd_service()
+        Log.debug(f"Start: Start {service} service")
+        self._execute.run_cmd(f"systemctl reset-failed {service}", check_error=False)
+        self._execute.run_cmd(f"systemctl start {service}", check_error=False)
+        while True:
+            Log.debug(f"Start: Starting {service} service")
+            status: str = self._get_service_status(service).strip()
+            if status == "active":
+                break
+            elif status == "failed":
+                Log.info(f"Start: Failed to start {service} and may cause failover or Stop.")
+                return const.OCF_ERR_GENERIC
+            else:
+                time.sleep(1)
+                continue
+        Log.info(f"Start: Started {service} service")
+        return const.OCF_SUCCESS
+
+    def stop(self) -> int:
+        """
+        Stop service. If stop failed it will cause stonith.
+
+        Returns:
+            int: Return as per service status.
+                if unknown or failed then return success.
+                timeout of stop will cause stonith.
+        """
+        service = self._get_systemd_service()
+        Log.debug(f"Stop: Stopping {service} service")
+        self._execute.run_cmd(f"systemctl stop {service}", check_error=False)
+        while True:
+            status: str = self._get_service_status(service).strip()
+            time.sleep(1)
+            if status in ["failed", "unknown"]:
```

Comment:
Do we know list of statuses of a service as displayed by systemd?

Reply:
i am not able to find all combination 
but found stable one are
**Stable status to take decision:**
failed: service failed
unknown: service stop
active: running
-----------------------------------------------------
**Unstable where process will wait till stable state arrive**
activating: stating
deactivating: stopping